consensus/e2e_tests/hotstuff_test.go

```diff
@@ -39,170 +39,22 @@ func TestHotstuff4Nodes1BlockHappyPath(t *testing.T) {
-		nodeState := GetConsensusNodeState(pocketNode)
-		assertNodeConsensusView(t, nodeId,
-			typesCons.ConsensusNodeState{
-				Height: startingHeight,
-				Step:   startingStep + 2,
-				Round:  uint8(startingRound),
-			},
-			nodeState)
-		require.Equal(t, leaderId, nodeState.LeaderId, fmt.Sprintf("%d should be the current leader", leaderId))
-	}
-
-	for _, message := range preCommitProposal {
-		P2PBroadcast(t, pocketNodes, message)
-	}
-	advanceTime(t, clockMock, 10*time.Millisecond)
-
-	// 4. Commit
-	preCommitVotes, err := WaitForNetworkConsensusEvents(t, clockMock, eventsChannel, consensus.PreCommit, consensus.Vote, numValidators, 500, true)
-	require.NoError(t, err)
-
-	for _, vote := range preCommitVotes {
-		P2PSend(t, leader, vote)
-	}
-	advanceTime(t, clockMock, 10*time.Millisecond)
-
-	commitProposal, err := WaitForNetworkConsensusEvents(t, clockMock, eventsChannel, consensus.Commit, consensus.Propose, numValidators, 500, true)
-	require.NoError(t, err)
-	for nodeId, pocketNode := range pocketNodes {
-		nodeState := GetConsensusNodeState(pocketNode)
-		assertNodeConsensusView(t, nodeId,
-			typesCons.ConsensusNodeState{
-				Height: startingHeight,
-				Step:   startingStep + 3,
-				Round:  uint8(startingRound),
-			},
-			nodeState)
-		require.Equal(t, leaderId, nodeState.LeaderId, fmt.Sprintf("%d should be the current leader", leaderId))
-	}
-
-	for _, message := range commitProposal {
-		P2PBroadcast(t, pocketNodes, message)
-	}
-	advanceTime(t, clockMock, 10*time.Millisecond)
-
-	// 5. Decide
-	commitVotes, err := WaitForNetworkConsensusEvents(t, clockMock, eventsChannel, consensus.Commit, consensus.Vote, numValidators, 500, true)
-	require.NoError(t, err)
-
-	for _, vote := range commitVotes {
-		P2PSend(t, leader, vote)
-	}
-	advanceTime(t, clockMock, 10*time.Millisecond)
-
-	decideProposal, err := WaitForNetworkConsensusEvents(t, clockMock, eventsChannel, consensus.Decide, consensus.Propose, numValidators, 500, true)
-	require.NoError(t, err)
-	for pocketId, pocketNode := range pocketNodes {
-		nodeState := GetConsensusNodeState(pocketNode)
-		// Leader has already committed the block and hence moved to the next height.
-		if pocketId == leaderId {
-			assertNodeConsensusView(t, pocketId,
-				typesCons.ConsensusNodeState{
-					Height: startingHeight + 1,
-					Step:   startingStep,
-					Round:  uint8(startingRound),
-				},
-				nodeState)
-			require.Equal(t, typesCons.NodeId(0), nodeState.LeaderId, "Leader should be empty")
-			continue
-		}
-		assertNodeConsensusView(t, pocketId,
-			typesCons.ConsensusNodeState{
-				Height: startingHeight,
-				Step:   startingStep + 4,
-				Round:  uint8(startingRound),
-			},
-			nodeState)
-		require.Equal(t, leaderId, nodeState.LeaderId, fmt.Sprintf("%d should be the current leader", leaderId))
-	}
+	decideProposal := waitForNextBlock(t, clockMock, eventsChannel, pocketNodes, startingHeight, startingStep, uint8(startingRound), numValidators, 500, true)
```

Comment:
Try reading this in isolation: `decideProposal := waitForNextBlock(`

It's very confusing. I expect `waitForNextBlock` to return a block, not the inside business logic of hotstuff.